module1/homework/make_unique/makeUnique.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <memory>
+#include <type_traits>
+
+namespace custom {
+template <typename Type, typename... Args, typename = std::enable_if_t<!std::is_array_v<Type>>>
+std::unique_ptr<Type> make_unique(Args&&... args) {
+    return std::unique_ptr<Type>(new Type(std::forward<Args>(args)...));
+}
+
+template <typename Type>
+std::unique_ptr<Type> make_unique(std::size_t size) {
+    return std::unique_ptr<Type>(new typename std::remove_extent<Type>::type[size]());
```

Comment:
Shouldnt you use Type[]?

Reply:
No. I put a comment like this in another reviews but it was my mistake. In fact array is automatically deduced and put instead of Type, eg. int[]. So it is already an array.

Reply:
ok, thanks :+1: